distributions.py

```diff
@@ -9,6 +9,331 @@ class ProbabilityDistribution:
+
+        Sets self.lbd and self.dim
+        """
+
+        # Situations: 
+        # 1. User specifies lbd as a list (disallow contradictory
+        #    dimension specification)
+        # 2. User specifies lbd as a list, then dim = len(lbd)
+        # 3. User specifies dim as a scalar, then lbd = np.ones(dim)
+        # 4. User specifies nothing, then dim = 1, lbd = 1
+
+        if lbd is not None and dim is not None:
+            if len(lbd) != dim:
+                raise ValueError('Input parameters lbd and dim must be the same dimension')
+            self.lbd = lbd
+            self.dim = dim
+
+        elif lbd is not None: # dim is None
+            dim = len(lbd)
+            self.lbd = lbd
+            self.dim = dim
+
+        elif dim is not None: # lbd is None, set lbd = 1
+            lbd = np.ones(dim)
+            self.lbd = lbd
+            self.dim = dim
+
+        else: # dim is None and lbd is None, dimension is one, lbd = 1
+            self.dim = 1
+            self.lbd = 1
+
+
+    def _convert_meanstdev_to_lbd(self, mean, stdev):
+        """
+        Converts user-given mean and stdev to an iterable lbd.
+        """
+
+        meaniter = isinstance(mean, (list, tuple))
+        stdviter = isinstance(stdev, (list, tuple))
+        lbd = []
+
+        # If they're both iterables:
+        if meaniter and stdviter:
+
+            # If one has length 1 and the other has length > 1:
+            if (len(mean) == 1) and (len(stdev) > 1):
+                mean = [mean for i in range(len(stdev))]
+            elif (len(stdev) == 1) and (len(mean) > 1):
+                stdev = [stdev for i in range(len(mean))]
+
+            for ind in range(len(mean)):
+                lb = self.meanstdev_to_lbd(mean[ind], stdev[ind])
+                lbd.append(lb)
+
+        # If mean is an iterable but stdev is not
+        elif meaniter:
+            for ind in range(len(mean)):
+                lb = self.meanstdev_to_lbd(mean[ind], stdev)
+                lbd.append(lb)
+
+        # If stdev is an iterable but mean is not
+        elif stdviter:
+            for ind in range(len(stdev)):
+                lb = self.meanstdev_to_lbd(mean, stdev[ind])
+                lbd.append(lb)
+
+        # If they're both scalars, let the following lbd checker cf vs dim
+        else:
+            lbd = self.meanstdev_to_lbd(mean, stdev)
+
+        return lbd
+
+
+    def _convert_lbd_to_iterable(self, lbd):
+        """
+        Converts user-input lbd to iterables. Ensures that the length
+        of the iterables matches on output.
+
+        If lbd is None, sets to 1.
+        """
+
+        # Tons of type/value checking for alpha/beta vs dim
+        lbditer = isinstance(lbd, (list, tuple))
+
+        if lbditer:
+            lbd = lbd
+        
+        elif lbd is None:
+            lbd = [1,]
+        
+        else: # lbd is a scalar
+            lbd = [lbd,]
+
+        return lbd
+
+    def meanstdev_to_lbd(self, mu, stdev):
```

Comment:
This looks to just make sure the parameters match an exponential distribution on [0, \infty). But we should allow users to build in affine maps. E.g., for any mu, stdev (>0), you can compute the affine map that would take that exponential distribution to the lbd=1 distribution on [0, \infty). 

There should also be some sort of flag to indicate whether the distribution is reflected or not. I.e., is it a distribution on (-\infty, 0] or [0, \infty)? Specifying `mu` and `stdev` is not enough to infer this, so we'll need an extra input.

Anyway, all of this would just be used to define the `AffineTransform`s in the constructor, and not much else.